### ISSUE TITLE
Inaccurate Git Status Decorations

### ISSUE THREAD

**Issue by jistjoalal** [microsoft/vscode#75112]
## Issue Type: <b>Bug</b>

**tl;dr** Git status decorations do not match changes on disk.

I am getting some strange behavior with git status decorations in the file explorer. Hopefully I am doing something dumb on my end that is easy to fix.

## Did not fix:
- Running Insiders
- Running with extensions disabled
- Restarting VSCode

## Steps to reproduce:
- Unstaged changes on master branch (nothing fancy here, I'm a git noob)
- Changes do not render in file explorer as status decorations

## More info

Here's what it looks like. `Timer` folder clearly has changes but none of the changed files have status decorations. `TimeData` folder has changes but is not even showing that as the case.

![pic](https://i.imgur.com/iVloIE3.png)

It is possible (although seems unlikely) that I messed up my repo somehow. Feel free to check on it [here](https://github.com/jistjoalal/cube-time). I will be pushing my current (buggy?) changes after filing this issue.

Any help is much appreciated! Seeing my changes at a glance is one of my favorite features. Thanks!

VS Code version: Code 1.35.0 (553cfb2c2205db5f15f3ee8395bbd5cf066d357d, 2019-06-04T01:17:12.481Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 1700 Eight-Core Processor           (16 x 3500)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.93GB (9.04GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-preview-github-styles|bie|0.1.6
vscode-eslint|dba|1.9.0
insert-iso-timestamp|dpk|0.0.1
prettier-vscode|esb|1.9.0
graphql-for-vscode|kum|1.13.2
vscode-duplicate|mrm|1.2.1
python|ms-|2019.5.18875
vscode-docker|Pet|0.6.2
vscode-graphql|Pri|0.1.8
vim|vsc|1.8.1
vscode-wakatime|Wak|2.1.2

</details>

**Comment by joaomoreno** [comment]
Can you easily reproduce this?

cc @jrieken

**Comment by jistjoalal** [comment]
Yes, atm I'm working around it by using the *Source Control* tab, which seems to be working OK.

*Source Control* tab is OK:

![screenshot1](https://i.imgur.com/pmHWyPC.png)

Here are some more screenshots of what's happening:

It looks like the parent directory is tracking the changes, but the individual file is not.

![screenshot](https://i.imgur.com/R3eq1p5.png)

And some more weirdness: a change from 4 commits ago is showing up in another file.

![screenshot2](https://i.imgur.com/6nnXiN9.png)

Again, the only changes to my repo atm are that one comment above.

![screenshot3](https://i.imgur.com/rS1MG9F.png)

Here is the link to that [commit](https://github.com/jistjoalal/cube-time/commit/619b101dd4505a26ca652b5fe1b3df0fbb49adb3#diff-0d0adfae9a44fd5376213e3854a470ed) (2 hours ago)

I will try to keep looking for patterns and maybe try working on another project to see if it happens over there as well. Thank you for the help!

**Comment by jistjoalal** [comment]
Yes, it was my bad changing my filenames.

Fixed the issue by [forcing git to be case-sensitive](https://stackoverflow.com/questions/17683458/how-do-i-commit-case-sensitive-only-filename-changes-in-git#20907647):
```bash
git config core.ignorecase false
```

It works!
![im dumb](https://i.imgur.com/lu0xFP4.png)

Hope I didn't waste anyone's time but my own 😅 closing the issue.
